### PR TITLE
Update return types

### PR DESCRIPTION
**Pull request by randytorres**

### CODE REVIEW

File: src/chains/eos_1_8/models/eosStructures.ts [hunk 1]
```diff
@@ -31,6 +31,11 @@ export type EOSGetTableRowsParams = {
   json: boolean
 }
 
+export type EosAuthorizationKeyStruct = {
```
Comment (traylewin) [inline src/chains/eos_1_8/models/eosStructures.ts:34]: Nice!

File: src/chains/eos_1_8/templates/chainActions/account_deleteAuth.ts [hunk 2]
```diff
@@ -35,11 +35,11 @@ export const decomposeAction = (action: EosActionStruct): DecomposeReturn => {
   const { name, data, authorization } = action
 
   if (name === actionName && data?.account && data?.permission) {
-    const auth = getAuthorization(authorization)
+    const auth = getFirstAuthorizationIfOnlyOneExists(authorization)
```
Comment (traylewin) [inline src/chains/eos_1_8/templates/chainActions/account_deleteAuth.ts:38]: you forgot the comments about why we're using getFirstAuthorizationIfOnlyOneExists - should be in every function using it

File: src/chains/eos_1_8/templates/chainActions/token_create.ts [hunk 3]
```diff
@@ -35,14 +35,14 @@ export const composeAction = ({
...
 export const decomposeAction = (action: EosActionStruct): DecomposeReturn => {
   const { name, data, account, authorization } = action
-  const auth = getAuthorization(authorization)
 
   if (name === actionName && data?.issuer && data?.maximum_supply) {
+    const auth = getFirstAuthorizationIfOnlyOneExists(authorization)
     const returnData: tokenCreateParams = {
       contractName: toEosEntityName(account),
-      ownerAccountName: toEosEntityName(auth.permission),
+      ownerAccountName: toEosEntityNameOrNull(auth?.permission),
```
Comment (traylewin) [inline src/chains/eos_1_8/templates/chainActions/token_create.ts:43]: is this supposed to be auth?.permission ??

File: src/chains/eos_1_8/templates/chainActions/account_deleteAuth.ts (hunk 2 above)
Comment (randytorres) [reply src/chains/eos_1_8/templates/chainActions/account_deleteAuth.ts:38]: hmm seems like there has to be a better way. I think its being used in 15+ places, do we really want the same comment everywhere?

Comment (traylewin) [reply src/chains/eos_1_8/templates/chainActions/account_deleteAuth.ts:38]: well the point of the comment is to make it clear why it's being called. A person looking at the code would likely be looking at just one of the functions which is why we need it in each one.

**Pull request merged**